### PR TITLE
[SPARK-15008][ML][PySpark] Add integration test for OneVsRest

### PR DESCRIPTION
**Pull request by yinxusen**
## What changes were proposed in this pull request?
1. Add `_transfer_param_map_to/from_java` for OneVsRest;
2. Add `_compare_params` in ml/tests.py to help compare params.
3. Add `test_onevsrest` as the integration test for OneVsRest.
## How was this patch tested?

Python unit test.

### CODE REVIEW

File: python/pyspark/ml/tests.py [hunk 1]
```diff
@@ -684,12 +684,32 @@ def test_logistic_regression(self):
         except OSError:
             pass
 
+    def _compare_params(self, m1, m2, param):
+        """
+        Compare 2 ML params, assert they have the same param. The param must be a parameter of m1.
```
Comment (jkbradley) [inline python/pyspark/ml/tests.py]: "Compare 2 ML params, assert they have the same param."
--> "Compare 2 ML Params instances for the given param, and assert both have the same param value and parent."

File: python/pyspark/ml/tests.py [hunk 2]
```diff
@@ -684,12 +684,32 @@ def test_logistic_regression(self):
...
+        """
+        # Prevent key not found error in case of some param neither in paramMap and
```
Comment (jkbradley) [inline python/pyspark/ml/tests.py]: "neither in paramMap and defaultParamMap" --> "in neither paramMap nor defaultParamMap"

File: python/pyspark/ml/tests.py [hunk 3]
```diff
@@ -772,6 +798,25 @@ def test_nested_pipeline_persistence(self):
             except OSError:
                 pass
 
+    def test_onevsrest(self):
```
Comment (jkbradley) [inline python/pyspark/ml/tests.py]: This is similar to the persistence test in OneVsRestTests.  Can you please remove the old one?

File: python/pyspark/ml/classification.py [hunk 4]
```diff
@@ -40,7 +41,7 @@
            'RandomForestClassifier', 'RandomForestClassificationModel',
            'NaiveBayes', 'NaiveBayesModel',
            'MultilayerPerceptronClassifier', 'MultilayerPerceptronClassificationModel',
-           'OneVsRest', 'OneVsRestModel']
+           'OneVsRest', 'OneVsRestModel', 'OneVsRestParams']
```
Comment (jkbradley) [inline python/pyspark/ml/classification.py]: OneVsRestParams should be private, so don't add it here.

File: python/pyspark/ml/classification.py [hunk 5]
```diff
@@ -1172,6 +1175,53 @@ def getClassifier(self):
         """
         return self.getOrDefault(self.classifier)
 
+    @classmethod
```
Comment (jkbradley) [inline python/pyspark/ml/classification.py]: None of these additions to OneVsRestParams are needed to make the tests run.  I thought they were only needed for tuning.py persistence.  Are they used anywhere?

**Comment by jkbradley** [comment]
@yinxusen Thanks!  I made an initial pass.

File: python/pyspark/ml/classification.py (hunk 5 above)
Comment (yinxusen) [inline python/pyspark/ml/classification.py]: I deleted them since they are only useful with previous save/load for tuning.

**Comment by jkbradley** [comment]
Sorry for the delay.  This LGTM once the merge conflicts are fixed and tests pass

**Comment by jkbradley** [comment]
Thanks for updating it.  In the merge, the old OneVsRest test was reinstated.  Could you please remove it again?

**Comment by yinxusen** [comment]
Sorry for the mistake. Fixed it.

**Comment by jkbradley** [comment]
No problem.  LGTM, but I'll rerun tests since there are many changes going on!

**Comment by jkbradley** [comment]
Merging with master and branch-2.0

**Pull request closed without merging**